Storage/RemoteTabs.swift

```diff
@@ -46,6 +49,8 @@ public protocol RemoteClientsAndTabs: SyncCommands {
     // Returns number of tabs inserted.
     func insertOrUpdateTabs(_ tabs: [RemoteTab]) -> Deferred<Maybe<Int>> // Insert into the local client.
     func insertOrUpdateTabsForClientGUID(_ clientGUID: String?, tabs: [RemoteTab]) -> Deferred<Maybe<Int>>
+
+    func deleteClientWithId(_ clientID: GUID) -> Success
```

Comment:
nit: `deleteClient(guid: GUID)` might be a more accurate way to describe this similar to how you have `getClient(guid: GUID)` above (to distinguish between `guid` vs `fxaDeviceId`